src/static_canvas.class.js

```diff
@@ -869,9 +865,16 @@
         fabric.util.clipContext(this, canvasToDrawOn);
       }
 
+      objsToRender = this._chooseObjectsToRender();
```

Comment:
Following your advice i grouped the big block of code and gave it a clear ( i hope ) name.